src/session_impl.cpp

```diff
@@ -5162,11 +5154,21 @@ namespace aux {
 		// proxy, and it's the same one as we're using for the tracker
 		// just tell the tracker the socks5 port we're listening on
 		if (m_socks_listen_socket && m_socks_listen_socket->is_open())
-			return m_socks_listen_socket->local_endpoint().port();
+		{
+			error_code ec;
+			tcp::endpoint endp = m_socks_listen_socket->local_endpoint(ec);
+			if (ec)
```

Comment:
I see. I wonder if that should be changed instead.. it seems like that would be nicer. I take it you build with exceptions disabled then? is it a problem to have naked throw expressions? should they be ``boost::throw_exception`` calls instead? if so, does boost.asio use that, or is that also a problem?